project/src/main/java/com/google/sps/data/UserPrefrences.java

```diff
@@ -0,0 +1,109 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.auto.value.AutoValue;
+import com.google.maps.model.LatLng;
+
+import java.util.List;
+
+/**
+ * Represents the prefrences entered by a user
+ */
+@AutoValue
+public abstract class UserPrefrences {
+
+  /**
+   * @return the minimal rating that the user wants to consider,
+   * represented by a number between 1-5.
+   */
+  public abstract float minRating();
+
+  /**
+   * @return the maximal range of prices the user wants to consider,
+   * represented by a number between 0-4.
+   */
+  public abstract int maxPriceLevel();
+
+  /**
+   * @return the coordinates of the location the user wants the food to be delivered to.
+  */
+  public abstract LatLng location();
+
+  /**
+  * @return the cuisine types that the user prefers.
+  */
+  public abstract List<String> cuisineTypes();
```

Comment:
Did you consider making this an immutable list?

Reply:
I didn't think about it, it does makes sense to make it immutable since it is for read-only uses. I changed it.